src/replication.c

```diff
@@ -831,32 +1024,78 @@ ask_replica_status_all(spec_t *spec)
 	MTX_UNLOCK(&spec->rq_mtx);
 }
 
-/*
- * This function processes the status response received from replica
- * and update spec's healthy/degrade count
- */
+static void
+update_rebuild_info(spec_t *spec, mgmt_ack_t *ack_data)
+{
+
+	uint64_t data_len = 0;
+
+	MTX_LOCK(&spec->rq_mtx);
+
+	if (spec->rebuild_info == NULL)
+		spec->rebuild_info = (mgmt_ack_t *)malloc(sizeof (mgmt_ack_t) *
+		    spec->rebuild_replica_count);
+
+	memcpy(&spec->rebuild_info[spec->rebuild_info_ack_recvd], ack_data,
+	    sizeof (mgmt_ack_t));
+	spec->rebuild_info_ack_recvd++;
+
+	if (spec->rebuild_info_ack_recvd == spec->rebuild_replica_count) {
+		data_len = spec->rebuild_info_ack_recvd * sizeof (mgmt_ack_t);
+		start_rebuild(spec->rebuild_info, spec->master_replica, data_len);
```

Comment:
master_replica, i.e., replica that need to be rebuilt, being gone need to be handled.

Reply:
also, error from start_rebuild need to be handled.

Reply:
Sir, replica being rebuilt and gone handled in handle_data_conn_error. 

if (spec->target_replica == r) {
}

With respect to handling error from start_rebuild, not sure what can we do. Do we want to go back to pick another replica ? If error came on fd, there should be a reason which might be caught through other code paths like dis-connect etc. No ?

Reply:
error from start_rebuild should restart the entire process of rebuild again.
disconnect path takes care of removing replica and its any pending IOs. But, here, error came on target_replica on which there are no pending mgmt_cmds.
Things like rebuild_in_progress etc need to be set to false.